programs/cryptid_signer/tests/003_execute_transaction.rs

```diff
@@ -0,0 +1,146 @@
+#![cfg(feature = "test-bpf")]
+
+mod util;
+
+use cryptid_signer::instruction::execute_transaction::ExecuteTransactionBuild;
+use cryptid_signer::instruction::expand_transaction::SeedOrAccount;
+use cryptid_signer::instruction::propose_transaction::ProposeTransactionBuild;
+use cryptid_signer::instruction::{CryptidInstruction, SigningKeyBuild};
+use cryptid_signer::state::InstructionData;
+use cryptid_signer::CryptidSignerSeeder;
+use dummy_program::DummyInstruction;
+use log::trace;
+use solana_generator::{build_instruction, system_program_id, PDAGenerator, SolanaAccountMeta};
+use solana_sdk::signer::Signer;
+use std::collections::HashMap;
+use std::iter::{empty, once};
+use util::*;
+
+#[tokio::test]
+async fn execute_transaction_test() {
```

Comment:
This test is quite large, is there any way to separate it into components, or move some code into libraries? Ideally, the test would read like a short paragraph with three components, given, when then, (or setup,test, evaluate). Maybe start with moving some setup into a context object or something like that.